File: ruby/src/Behavioral/Template_Method/GameTitleInfo.rb
#//GameTitleInfo - three of three concrete templates

require "Behavioral/Template_Method/TitleInfo"

class GameTitleInfo < TitleInfo
	def initialize(titleName)
		setTitleName(titleName)
	end

	def getTitleBlurb ()
		"Game: #{getTitleName()}"
	end

end

File: csharp/src/Behavioral/Template_Method/GameTitleInfo.cs
namespace Behavioral.Template_Method {


//GameTitleInfo.java - three of three concrete templates

public class GameTitleInfo : TitleInfo {  
   public GameTitleInfo(string titleName) {
       this.setTitleName(titleName);
   }     
   
   override public string getTitleBlurb() {
       return ("Game: " + this.getTitleName());
   }
}

}
